code/obj/item/device/pocketwatch.dm

```diff
@@ -0,0 +1,182 @@
+/obj/item/pocketwatch
+	name = "T-MO Watch"
+	desc = "A high-tech pocketwatch with an electronic ink display. Hooked into the station's monitoring systems, it can provide personnel with important financial information instantly from far away, and also tell the time with a faux-analog display. Time is money!"
+	icon = 'icons/obj/objects.dmi'
+	inhand_image_icon = 'icons/mob/inhand/hand_tools.dmi'
+	icon_state = "watch_closed"
+	item_state = "pocketwatch"
+	force = 0
+	throwforce = 0
+	w_class = W_CLASS_SMALL
+	object_flags = NO_GHOSTCRITTER
+	inventory_counter_enabled = TRUE
+	var/current_clock_mode = "Time Keeping"
+	var/hour_minute_divider = ":" //used to make the colon in the time blink like a clock display
+	var/display_type = "clock" //used to show analog or screen displays on sprite
+	var/text_to_display
+	var/emagged = FALSE
+
+	var/steps_taken = 0
+	var/counting_steps = FALSE //true when in hands or pockets
+
+	var/currently_timing = FALSE //used for timer mode
+	var/timer_time = 0
+	var/last_tick
+
+	var/bell_counter = 0
+	var/image/alert_overlay
+
+	var/list/clock_modes = list("Time Keeping",
+	"Total Station Budget Monitor",
+	"Payroll Budget Monitor",
+	"Cargo Budget Monitor",
+	"Research Budget Monitor",
+	"Total PTL Net Income",
+	"Step Counter",
+	"Service Bell Ring Counter",
+	"Timer")
+
+	New()
+		processing_items.Add(src)
+		create_inventory_counter()
+		if(istype(loc, /mob/)) //start counting steps if we spawn in the HoP's pocket!!
+			start_counting_steps(loc)
+		..()
+
+	proc/format_time_text(var/timeValue as num)
+		var/seconds = round((timeValue / 10) % 60)
+		var/minutes = round(((timeValue / 10) - seconds) / 60)
+		if (minutes < 10)
+			minutes = "0[minutes]"
+		if (seconds < 10)
+			seconds = "0[seconds]"
+		return "[minutes]:[seconds]"
```

Comment:
since this is stolen from game clocks it should probably just go in `helpers.dm`